paper-autocomplete.js

```diff
@@ -652,15 +648,17 @@ Polymer({
    * @returns {Boolean}
    */
 	validate() {
-		return this.$.autocompleteInput.validate();
+		const valid = this.$.autocompleteInput.validate();
+		if (this.restricted && valid) {
+			return !(this.$.autocompleteInput.invalid = !this.value);
```

Comment:
Please expand this a bit for clarity.
```suggestion
			const hasValue = this.value != null;
			this.$.autocompleteInput.invalid = !hasValue;
			return hasValue;
```

Reply:
This could also be empty text or maybe undefined (not sure).

Reply:
Well the check handles undefined, but it's a bit messy if it's empty text since it's "declared" as an Object.